### ISSUE TITLE
Not working in macOS 11 beta

### ISSUE THREAD

**Issue by sweettimes** [justinfrankel/licecap#92]
on macOS Big Sur 11 Beta, it just shows the frame, and the results after recording is totally black

![Screen Shot 2020-07-31 at 10 45 29](https://user-images.githubusercontent.com/566289/88997877-ffa78300-d31a-11ea-933c-b9e95f984614.png)

**Comment by rstockm** [comment]
I can confirm this for the Developer Betas 1 to 4. REAPER is running almost flawfless though, with one exception: the old Theme development/tweaker  windows:

<img width="573" alt="image" src="https://user-images.githubusercontent.com/3195116/90384544-01df4080-e082-11ea-8f5a-a999c5067218.png">

Maybe these two are connected?

**Comment by jjozsi** [comment]
I can confirm it's still the same with Big Sur 11.0.1 Release Candidate 1

**Comment by 4N7** [comment]
I've also confirmed this bug for `Big Sur 11.0.1 (20B29)`.
Permission to record screen have been granted, as shown in attached image.

![img](https://user-images.githubusercontent.com/27744918/99193667-793b7380-2748-11eb-9865-9c75062d0b05.png)

**Comment by kenji21** [comment]
Clone the project (needed to change build setting for c++ `CLANG_CXX_LIBRARY = "libc++";`, and minimal version to macOS 10.9 because of Xcode 12)

Capture seems to work:
![capture](https://user-images.githubusercontent.com/1105089/100520180-bf5af300-319c-11eb-9b07-5122f7e8498f.gif)

Captured View Hierarchy to determine which view is having the window not transparent:
<img width="1131" alt="Screenshot 4" src="https://user-images.githubusercontent.com/1105089/100520243-152f9b00-319d-11eb-9996-f864fbf8b707.png">

The [NSVisualEffectView](https://developer.apple.com/documentation/appkit/nsvisualeffectview) below the SWELL_APP_PREFIX_hwnd is not opaque

Tried to make it [inactive](https://developer.apple.com/documentation/appkit/nsvisualeffectstate/nsvisualeffectstateinactive?language=objc):
```
(lldb) expr -- [((NSVisualEffectView *) 0x7fa4c172df90) setState:2]
```
without success

**Comment by mzl980425** [comment]
+1

**Comment by guoxiangke** [comment]
+1

**Comment by esaruoho** [comment]
+1 - please would it be possible to make LiceCap work on macOS Big Sur and M1 processors?

**Comment by esaruoho** [comment]
@fregante what's with the thumbdowns ;)

**Comment by fregante** [comment]
> ### Stop spamming with "me too" comments. 
> 
> Of course it doesn't work for anyone. Comments will not make the fix appear any faster.

**Comment by esaruoho** [comment]
> without success

Ok, I'll try that:
![Screenshot 2021-01-07 at 11 22 58](https://user-images.githubusercontent.com/4966687/103875161-c26dcb00-50da-11eb-9f21-fbbd021d2626.png)

After accepting the changes, I clicked on macOS Deployment Target 111.
There's still 5 issues, which are:
![Screenshot 2021-01-07 at 11 24 06](https://user-images.githubusercontent.com/4966687/103875249-dc0f1280-50da-11eb-8293-bca7b0181c55.png)

so i start with migrating the four (french, english, german, japanese). will press Migrate on everything:
![Screenshot 2021-01-07 at 11 24 26](https://user-images.githubusercontent.com/4966687/103875289-e7623e00-50da-11eb-8f6a-f079ed78cd3f.png)

German Japanese went quickly, but English says stuff like this:

![Screenshot 2021-01-07 at 11 24 54](https://user-images.githubusercontent.com/4966687/103875331-f812b400-50da-11eb-91cc-a0caf4876e5e.png)

Since I'm working in the blind, I'll just press Migrate.

And the last one which is Enable Base Internationalization - again, working in the blind, clicking Finish.
![Screenshot 2021-01-07 at 11 25 27](https://user-images.githubusercontent.com/4966687/103875400-0bbe1a80-50db-11eb-90f6-722992e2a618.png)

Now the topbar does not show any warnings or errors. Pressing Build. After it, 17 warnings and 2 errors - with build failed. 
Looking at what else to fix here that would get rid of some of the warnings.

![Screenshot 2021-01-07 at 11 26 33](https://user-images.githubusercontent.com/4966687/103875508-33ad7e00-50db-11eb-8092-9b9225e9b321.png)

now changed Apple Clang - Language - C++ to libc++.
![Screenshot 2021-01-07 at 11 27 36](https://user-images.githubusercontent.com/4966687/103875638-5b9ce180-50db-11eb-9d6d-6adadfc082dd.png)

that reduced warnings from 17 to 13, and 2 errors still exist.

**Comment by esaruoho** [comment]
after running through the possible misuse of comma operators and fixing them, i got this far:
6 warnings, 2 errors:
![Screenshot 2021-01-07 at 12 37 11](https://user-images.githubusercontent.com/4966687/103882846-11206280-50e5-11eb-8f75-d3046bf2b7f7.png)
not entirely sure how to proceed. i did add Metal framework, but did not change OpenGL for anything.

Feels like the Value Conversion issues are new ones caused by the "comma operator" fixes. but also feels like stat64 is a huge thing.

Not sure how to change the "Couldn't load spec with identifier 'com.intel.compilers.icc.11_1_0' in domain 'macosx'" -> should it maybe be macOS?

**Comment by esaruoho** [comment]
> Clone the project (needed to change build setting for c++ `CLANG_CXX_LIBRARY = "libc++";`, and minimal version to macOS 10.9 because of Xcode 12)

hi @kenji21 - can you specify where you changed CLANG_CXX_LIBRARY to libc++, please?
I tried some changes, but i'm not sure if i got them all.

**Comment by wojosc** [comment]
Thank you for working on this. I also have the same problem with the licecap window on macOS Big Sur showing just grey.

**Comment by matt** [comment]
Hi Friends. I finally updated to Big Sur today...

First of all, thank you to @kenji21 for identifying steps 2 and 3 below. :raised_hands: Please understand, this is not an ideal solution, but rather a quick fix with minimal effort.

Quick and _**dirty**_ solution:

MacOS 11.2
Xcode 12.4
Intel (Please post whether this is sufficient for Arm as well; I am not able to test at the moment)

Step 1: Clone the project

Step 2: Update Deployment Target 
<img width="1359" alt="Screen Shot 2021-02-03 at 10 58 35 PM" src="https://user-images.githubusercontent.com/2709/106843341-d2de7a80-6673-11eb-8952-43919cd4618f.png">

Step 3: Update C++ Standard Library
<img width="1359" alt="Screen Shot 2021-02-03 at 10 58 27 PM" src="https://user-images.githubusercontent.com/2709/106843294-b6dad900-6673-11eb-91ba-be3817ceb542.png">

Step 4: Set a clear background on the SWELL_APP_PREFIX_modelesswindow
`[self setBackgroundColor:NSColor.clearColor];`
<img width="1359" alt="Screen Shot 2021-02-03 at 11 05 37 PM" src="https://user-images.githubusercontent.com/2709/106843611-6152fc00-6674-11eb-97bf-2f98841ed4cb.png">

Step 5: Fill the RECT for SWELL_APP_PREFIX_hwnd with a color of your choosing (this will be the border color)
```
[[NSColor systemGrayColor] set];
NSRectFill(NSMakeRect(0,0,r.right+r.left,r.bottom+r.top+20));
```
<img width="1359" alt="Screen Shot 2021-02-03 at 11 05 16 PM" src="https://user-images.githubusercontent.com/2709/106843609-5f893880-6674-11eb-8b3b-c79658344cc8.png">


cc: @sweettimes @rstockm @jjozsi @4N7 @kenji21 @alivevietnam @mzl980425 @esaruoho @fregante @wojosc

**Comment by esaruoho** [comment]
> Intel (Please post whether this is sufficient for Arm as well; I am not able to test at the moment)

Hi, I tried your stuff with M1, but unfortunately am still seeing issues such as the following:
<img width="400" alt="Screenshot 2021-02-04 at 14 09 31" src="https://user-images.githubusercontent.com/4966687/106890871-9b62e300-66f2-11eb-8974-a77b6454b59d.png">

Not sure how to solve them. But thanks for chiming in! BTW, took a while to figure out when you modified something, and when you added something, and whcih filenames were the files in question. Otherwise, was able to figure it out out of the attached screenshots, so I do appreciate the work involved for sure!

**Comment by noestreich** [comment]
Hi @matt, 
any chance you could provide a binary?

**Comment by logananderson** [comment]
Hey folks, really needed Licecap this morning for a presentation so I followed @matt's instructions.  Here's a fork if it helps anyone: https://github.com/logananderson/licecap

Working for me on Big Sur 11.0.1 (20B50) on a MacBook Pro (Retina, 15-inch, Mid 2015).

**Comment by esaruoho** [comment]
> Working for me on Big Sur 11.0.1 (20B50) on a MacBook Pro (Retina, 15-inch, Mid 2015).

Interesting, let's see how far I get with this fork and M1. Hopefully less errors? Did you update WDL?

**Comment by esaruoho** [comment]
Ok, unfortunately even with the @logananderson fork, I get all these after trying to build:
<img width="626" alt="Screenshot 2021-02-04 at 22 59 23" src="https://user-images.githubusercontent.com/4966687/106954570-ab9fb000-673c-11eb-8bb5-84ce711b9297.png">
<img width="654" alt="Screenshot 2021-02-04 at 22 59 55" src="https://user-images.githubusercontent.com/4966687/106954595-b5c1ae80-673c-11eb-9ca8-a9a78c9c3187.png">

**Comment by ganjarpanji** [comment]
for anyone who need the latest app, dyou can download here https://drive.google.com/file/d/1zJjtR0Ba23zRW1cWV7kXs27gGPKv_4lr/view?usp=sharing

**Comment by rstockm** [comment]
> for anyone who need the latest app, dyou can download here https://drive.google.com/file/d/1zJjtR0Ba23zRW1cWV7kXs27gGPKv_4lr/view?usp=sharing

Thank you so much. This actually works under Rosetta2/ARM on a M1!
One Bug though: the title screen appears only for a split-second no matter what time I set. But that's a minor complain.

**Comment by esaruoho** [comment]
yay, i was able to install and run it on my Big Sur 11.2.1 and M1. Works as expected. I hope that #101  will be merged in by @justinfrankel  soon.

**Comment by itsthomas** [comment]
> for anyone who need the latest app, dyou can download here https://drive.google.com/file/d/1zJjtR0Ba23zRW1cWV7kXs27gGPKv_4lr/view?usp=sharing

Thanks mate. I confirm that it's also working on Intel Macs.

**Comment by dialex** [comment]
Fixed in the latest version Licecap 1.31 (this commit https://github.com/justinfrankel/licecap/commit/641c098841f8d85bd871332b6b4ae9510022f986) 🎉
You can close this issue @sweettimes
